### PR TITLE
Fixes Bounty #Generate equirectangular 360 panorama from babylon.js scene

### PR DESCRIPTION
**Pull request by lokiiarora**
Hey, I'm a first-time contributor to Babylon.js, love the community, and would love it if someone could review this PR.
This PR is based on @Pryme8 's work on this playground [here](https://playground.babylonjs.com/#JTUAY3#5).

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14251/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14251/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14251/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14251/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14251/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14251/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14251/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14251/merge#BCU1XR#0

**Comment by lokiiarora** [comment]
> That's really great! thank you so much for that :-)
> 
> Got a question. I tried it, and it works great. The only thing I noticed - the image generated is flipped:
> 
> https://playground.babylonjs.com/?snapshot=refs/pull/14251/merge#Z6SWJU#5
> 
> results in:
> 
> equirectangular_capture (1)
> 
> Was that intentional?

Ouch, this was not intentional, I guess I'm not passing `invertY` flag, let me just do it

**Comment by lokiiarora** [comment]
@RaananW fixed on local, will verify on remote :)

**Comment by lokiiarora** [comment]
looks fine now @RaananW

File: packages/dev/core/src/Misc/equirectangularCapture.ts
Comment (RaananW) [inline packages/dev/core/src/Misc/equirectangularCapture.ts:100]: Would be cleaner to move this to the shader directory. You can then import the shader code here and it will be automatically populated (and minified a bit...)

**Review by RaananW — COMMENTED** [review]
LGTM! just a note about the shader code itself. I'll also wait for others to comment, but I will approve right after the shader code moves.

File: packages/dev/core/src/Misc/equirectangularCapture.ts
Comment (lokiiarora) [reply packages/dev/core/src/Misc/equirectangularCapture.ts]: this is done on the latest commit

**Review by RaananW — APPROVED** [review]

**Review by carolhmj — APPROVED** [review]
LGTM 😸

File: packages/dev/core/src/Misc/equirectangularCapture.ts
Comment (Popov72) [inline packages/dev/core/src/Misc/equirectangularCapture.ts:51]: It should be `equirectangularPanorama` and you should rename the file to be `equirectangularPanorama.fragment.fx`, to comply with our naming rules.

**Review by Popov72 — CHANGES_REQUESTED** [review]

**Review by sebavan — COMMENTED** [review]
Aside of naming, this looks great to me !!!

**Review by sebavan — APPROVED** [review]
Could you allow an option to pass the probe in ?

File: packages/dev/core/src/Misc/equirectangularCapture.ts
Comment (lokiiarora) [reply packages/dev/core/src/Misc/equirectangularCapture.ts]: added the correct naming scheme @Popov72

**Comment by lokiiarora** [comment]
> Could you allow an option to pass the probe in ?

has been added, do check if this is how you would want it to be?

File: packages/dev/core/src/Misc/equirectangularCapture.ts
Comment (Popov72) [inline packages/dev/core/src/Misc/equirectangularCapture.ts:36]: nit: size and position are ignored

Comment (lokiiarora) [reply packages/dev/core/src/Misc/equirectangularCapture.ts]: added it

**Review by Popov72 — APPROVED** [review]

**Comment by sebavan** [comment]
Nice, looks great !!!!

**Comment by sebavan** [comment]
I do not think it is supported  at the moment. It would be great if you could create a topic in the forum with a repro in the playground. This way we could simply check how hard this could be to support.

**Pull request merged**
